### PR TITLE
Divergent iterations setter for HMC and Monomial gamma

### PR DESCRIPTION
**Pull request by ben18785**
Sorts #679

### CODE REVIEW

File: pints/_mcmc/_hamiltonian.py [hunk 1]
```diff
@@ -102,6 +102,23 @@ def set_epsilon(self, epsilon):
         self._epsilon = epsilon
         self._set_scaled_epsilon()
 
+    def set_hamiltonian_threshold(self, hamiltonian_threshold):
```
Comment (MichaelClerx) [inline pints/_mcmc/_hamiltonian.py]: Alphabetise?

File: pints/tests/test_mcmc_hamiltonian.py [hunk 2]
```diff
@@ -138,6 +138,19 @@ def test_set_hyper_parameters(self):
         self.assertEqual(mcmc.leapfrog_step_size()[0], 1.5)
         self.assertEqual(mcmc.leapfrog_step_size()[1], 3)
 
+    def test_other_setters(self):
+        """
```
Comment (MichaelClerx) [inline pints/tests/test_mcmc_hamiltonian.py]: For a stupid technical reason these shouldn't be docstrings but just inline comments ```# These guys```

File: pints/tests/test_mcmc_monomial_gamma_hmc.py [hunk 3]
```diff
@@ -179,6 +179,19 @@ def test_set_hyper_parameters(self):
         self.assertEqual(mcmc.mass(), m)
         self.assertRaises(ValueError, mcmc.set_mass, -1.8)
 
+    def test_other_setters(self):
+        """
```
Comment (MichaelClerx) [inline pints/tests/test_mcmc_monomial_gamma_hmc.py]: As above

File: pints/_mcmc/_hamiltonian.py (hunk 1 above)
Comment (ben18785) [reply pints/_mcmc/_hamiltonian.py]: Done

File: pints/tests/test_mcmc_hamiltonian.py (hunk 2 above)
Comment (ben18785) [reply pints/tests/test_mcmc_hamiltonian.py]: Done

File: pints/tests/test_mcmc_monomial_gamma_hmc.py (hunk 3 above)
Comment (ben18785) [reply pints/tests/test_mcmc_monomial_gamma_hmc.py]: Done

**Pull request merged**